### PR TITLE
chore: upgrading sampler to v08 [LIT-783]

### PR DESCRIPTION
**Pull request by SavDont**
# Description

Upgrading sampler solidity contracts to use v08 solidity contracts.

# Testing & Simbot Run

[Mainnet](https://metabase.spaceship.0x.org/dashboard/186?run_id=sampler-solidityV08-upgrade-p3&adjusted_amount_bought_win_tolerance__bps_=1.0)

# Checklist

-   [ ] Update 0x API documentation (gitbook) if needed (e.g. public facing API change).
-   [ ] Add tests to cover changes as needed.
-   [ ] All dependent changes (e.g. RFQ server, Gas Price Oracle) have been deployed (if any).

### CODE REVIEW

File: contracts/src/NativeOrderSampler.sol [hunk 1]
```diff
@@ -191,7 +189,7 @@ contract NativeOrderSampler {
         if (
             orderInfo.status != IExchange.OrderStatus.FILLABLE ||
             !isSignatureValid ||
-            order.makerToken == IERC20TokenV06(0)
+            address(order.makerToken) == address(0)
```
Comment (dextracker) [inline contracts/src/NativeOrderSampler.sol:192]: might want to change this to IERC20TokenV08 just to keep it consistent

File: contracts/src/interfaces/IGMX.sol [hunk 2]
```diff
@@ -1,5 +1,5 @@
-pragma solidity ^0.6;
-pragma experimental ABIEncoderV2;
+pragma solidity ^0.8;
+pragma experimental ABIEncoderV2; //TODO: remove this
```
Comment (dextracker) [inline contracts/src/interfaces/IGMX.sol:2]: i think ABIEncoderV2 can be dropped

**Review by dextracker — APPROVED** [review]
LGTM just a few nitpicks

**Review by kyu-c — APPROVED** [review]

**Comment by kyu-c** [comment]
There is a quite meaningful difference in the simbot result. Any idea why that is the case?

**Comment by SavDont** [comment]
> There is a quite meaningful difference in the simbot result. Any idea why that is the case?

I'm not too sure but could it potentially be due to routing through 0x liquidity which dev doesn't have access to?

**Comment by kyu-c** [comment]
> > There is a quite meaningful difference in the simbot result. Any idea why that is the case?
> 
> I'm not too sure but could it potentially be due to routing through 0x liquidity which dev doesn't have access to?

Ah, yeah that seems the reason.

Running with `excludedSources=0x` can be useful for testing changes like this (no need to re-run it again as long as you feel confident).

**Comment by SavDont** [comment]
Closing this for now, we're currently researching another solution for conflicting sampler contract version using overrides. Therefore, v08 contracts will likely not be needed as part of sampler.

**Pull request closed without merging**